kani-compiler/src/kani_middle/transform/check_uninit/delayed_ub/instrumentation_visitor.rs

```diff
@@ -18,20 +18,17 @@ use rustc_middle::ty::TyCtxt;
 use stable_mir::mir::{
     mono::Instance,
     visit::{Location, PlaceContext},
-    BasicBlockIdx, MirVisitor, Operand, Place, Rvalue, Statement, Terminator,
+    MirVisitor, Operand, Place, Rvalue, Statement, Terminator,
 };
 use std::collections::HashSet;
 
 pub struct InstrumentationVisitor<'a, 'tcx> {
-    /// Whether we should skip the next instruction, since it might've been instrumented already.
-    /// When we instrument an instruction, we partition the basic block, and the instruction that
-    /// may trigger UB becomes the first instruction of the basic block, which we need to skip
-    /// later.
-    skip_next: bool,
-    /// The instruction being visited at a given point.
-    current: SourceInstruction,
-    /// The target instruction that should be verified.
-    pub target: Option<InitRelevantInstruction>,
+    /// All target instructions in the body.
+    targets: Vec<InitRelevantInstruction>,
+    /// Currently analyzed instruction.
+    current_instruction: SourceInstruction,
```

Comment:
This seems to be a duplicate of `current_target.source`... what am I missing?

Reply:
My intention was to use `current_instruction` as a counter, which is changed while iterating over basic blocks and `current_target.source` to be a part of the instrumentation target. The difference between two was supposed to indicate that we have already switched to the next statement/terminator. 

I agree though that `current_instruction` seems redundant. I will refactor it slightly to use `current_target.source` for both purposes.